src/Microsoft.AspNetCore.Blazor.Build/targets/RazorCompilation.targets

```diff
@@ -19,8 +19,10 @@
   <!-- Something quick for input/output tracking - the assumptions here should match what the CLI does -->
   <ItemGroup>
     <BlazorGenerate Include="**\*.cshtml" />
-    <Content Update="@(Content->WithMetadataValue('Extension', '.cshtml'))">
+    <_BlazorGenerateDeclarationContent Include="@(Content->WithMetadataValue('Extension', '.cshtml'))" />
+    <Content Update="@(_BlazorGenerateDeclarationContent)">
       <Generator>MSBuild:BlazorGenerateDeclaration</Generator>
+      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
```

Comment:
Its weird to me that this didn't do the right thing. I believe you that there's a bug here, but do you know the explanation?

Is the general idea that `Update="..."` doesn't do what you want unless the you're passing it an actual, unmodified itemgroup?

Reply:
It's weird to me too, but it didn't do the right thing before. At least, MSBuild Structured Log Viewer listed all the non-cshtml files with the `Generator: MSBuild:BlazorGenerateDeclaration` metadata as well as the cshtml files.

Since I can't find a single other example of anyone on the internet combining `Update` with `->WithMetadataValue`, it's hard to say whether this syntax is meant to be legit or not.